src/submission/validation-clinical/therapy.ts

```diff
@@ -270,3 +293,41 @@ export function getTreatment(
 
 	return treatment;
 }
+
+const validateDrugTreatmentFields = (
+	therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+	errors: SubmissionValidationError[],
+) => {
+	const { drug_name, drug_rxnormcui, drug_database, drug_term, drug_id } = therapyRecord;
+	const isRxNorm = Boolean(drug_name && drug_rxnormcui);
+	const isDrugDb = Boolean(drug_database && drug_id && drug_term);
+
+	if (!isRxNorm && isDrugDb) {
+		// No RxNorm, All Drug DB Fields, no further validation
+		return;
+	} else if (!isRxNorm && !isDrugDb) {
+		// Not RxNorm, but Not All Drug Fields -> Error
+		const drugFields = { drug_database, drug_term, drug_id };
+		const missingFields: ClinicalInfo[] = [];
+		for (const [drugKey, drugValue] of Object.entries(drugFields)) {
+			if (!drugValue) {
+				const data = { [drugKey]: drugValue };
+				missingFields.push(data);
+			}
+		}
+
+		missingFields.forEach((field) => {
+			const [drugKey] = Object.entries(field)[0];
+			const fieldName = drugKey as ClinicalFields;
```

Comment:
Not 100% clear why `missingField` is getting an object if the `drugValue` is falsy
Later on the `missingFields` iterator, `drugValue` isn't used.
`drugValue` could still be a string "" or number 0 and be falsey

Reply:
https://github.com/icgc-argo/argo-clinical/pull/1195/files#diff-57392d148ba45d8530fa1789680e51f9398e3b20e910bccf1fa0e95b78445a08R314

Reply:
I think both comments can be addressed by condensing the logic: [Condense validation](https://github.com/icgc-argo/argo-clinical/pull/1195/commits/f0c62955ccdf93fccdc43a5026f9088613b213ab)
(passes unit tests)

This is just a block that was not optimized and cleaned up. Extra verbose logic from working out the solution.